homeassistant/components/sensor/pollen.py

```diff
@@ -0,0 +1,349 @@
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit
+
+
+class AllergyAverageSensor(BaseSensor):
+    """Define a sensor to show allergy average information."""
+
+    def update(self):
+        """Update the status of the sensor."""
+        self.data.update()
+
+        data_attr = getattr(self.data, self._data_params['data_attr'])
+        indices = [
+            p['Index']
+            for p in data_attr['Location']['periods']
+        ]
+        average = average_of_list(indices)
+
+        self._attrs[ATTR_TREND] = calculate_trend(indices)
+        self._attrs[ATTR_CITY] = data_attr['Location']['City'].title()
+        self._attrs[ATTR_STATE] = data_attr['Location']['State']
+        self._attrs[ATTR_ZIP_CODE] = data_attr['Location']['ZIP']
+
+        [rating] = [
+            i['label'] for i in RATING_MAPPING
+            if i['minimum'] <= average <= i['maximum']
+        ]
+        self._attrs[ATTR_RATING] = rating
+
+        self._state = average
+        self._unit = 'index'
+
+
+class AllergyIndexSensor(BaseSensor):
+    """Define a sensor to show allergy index information."""
+
+    def update(self):
+        """Update the status of the sensor."""
+        self.data.update()
+
+        location_data = self.data.current_data['Location']
+        [period] = [
+            p for p in location_data['periods']
+            if p['Type'] == self._data_params['key']
+        ]
+
+        self._attrs[ATTR_ALLERGEN_GENUS] = period['Triggers'][0]['Genus']
+        self._attrs[ATTR_ALLERGEN_NAME] = period['Triggers'][0]['Name']
+        self._attrs[ATTR_ALLERGEN_TYPE] = period['Triggers'][0]['PlantType']
+        self._attrs[ATTR_OUTLOOK] = self.data.outlook_data['Outlook']
+        self._attrs[ATTR_SEASON] = self.data.outlook_data['Season']
+        self._attrs[ATTR_TREND] = self.data.outlook_data[
+            'Trend'].title()
+        self._attrs[ATTR_CITY] = location_data['City'].title()
+        self._attrs[ATTR_STATE] = location_data['State']
+        self._attrs[ATTR_ZIP_CODE] = location_data['ZIP']
+
+        [rating] = [
+            i['label'] for i in RATING_MAPPING
+            if i['minimum'] <= period['Index'] <= i['maximum']
+        ]
+        self._attrs[ATTR_RATING] = rating
+
+        self._state = period['Index']
+        self._unit = 'index'
+
+
+class AllergyAveragesData(object):
+    """Define an object to averages on future and historical allergy data."""
+
+    def __init__(self, client):
+        """Initialize."""
+        self._client = client
+        self.extended_data = None
+        self.historic_data = None
+
+    @Throttle(MIN_TIME_UPDATE_AVERAGES)
+    def update(self):
+        """Update with new data."""
+        from pypollencom.exceptions import HTTPError
+
+        try:
+            self.extended_data = self._client.allergens.extended()
```

Comment:
You could break out a function to access a client method inside a try except block, since you do the same thing 6 times.